packages/noya-state/src/reducers/exportReducer.ts

```diff
@@ -0,0 +1,113 @@
+import Sketch from '@sketch-hq/sketch-file-format-ts';
+import produce from 'immer';
+import { getSelectedLayers } from '../selectors/layerSelectors';
+import { ApplicationState } from './applicationReducer';
+
+export type ExportAction =
+  | [type: 'setScale', index: number, value: string]
+  | [type: 'setName', index: number, value: string]
+  | [type: 'setFileFormat', index: number, value: Sketch.ExportFileFormat]
+  | [
+      type: 'setNamingScheme',
+      index: number,
+      value: Sketch.ExportFormatNamingScheme,
+    ]
+  | [type: 'addExportFormat']
+  | [type: 'deleteExportFormat', index: number];
+
+export function exportReducer(
+  state: ApplicationState,
```

Comment:
Can we pass just the `ExportOptions` type here? That'll make everything a little simpler I think. We can call the reducer from within `applicationReducer` for each layer.

Similar to https://github.com/noya-app/noya/blob/5380076b51cc8e225b854cc9fecbd7f2ae751450/packages/noya-state/src/reducers/applicationReducer.ts#L147-L158